trade-finance-manager-ui/server/controllers/case/parties/index.test.js

```diff
@@ -1269,33 +1708,75 @@ describe('controllers - case - parties', () => {
               _id: '1',
             },
             session,
+            url: `/case/1/parties/${party}`,
           };
 
-          await partiesController.postBuyerPartyDetails(req, res);
+          await partiesController.postPartyDetails(req, res);
           expect(res.redirect).toHaveBeenCalledWith('/not-found');
         });
       });
 
-      describe('when user is not allowed to edit', () => {
-        it('should redirect to not-found route', async () => {
-          const req = {
-            body: {
-              csrf: '12345',
-            },
-            params: {
-              _id: '1',
-            },
-            session: {
-              user: userNotAllowedToEdit,
+      describe('when deal exists', () => {
+        const mockDeal = {
+          _id: '61f6ac5b02ffda01b1e8efef',
+          mock: true,
+          tfm: {
+            parties: {
+              indemnifier: {
+                partyUrnRequired: true,
+              },
             },
-          };
+          },
+        };
+
+        const req = {
+          params: {
+            _id: mockDeal._id,
+            urn: '1234',
+          },
+          body: {
+            csrf: '12345',
+          },
+          session,
+          url: `/case/1/parties/${party}`,
+        };
+
+        beforeEach(() => {
+          api.getDeal = () => Promise.resolve(mockDeal);
+          api.getAmendmentsByDealId = () => Promise.resolve({ data: [] });
+          api.getParty = () => Promise.resolve({
+            status: 200,
+            data: mockCompany,
+          });
+          api.updateParty = () => jest.fn();
+        });
+
+        it('should render error page when party type is invalid', async () => {
+          req.body.url = '/';
+
+          expect(res.redirect).toHaveBeenCalledWith('/not-found');
+        });
+
+        it('should render error page when party URN is empty', async () => {
+          req.body.url = `/case/1/parties/${party}`;
+          req.params.urn = '';
 
-          await partiesController.postBuyerPartyDetails(req, res);
           expect(res.redirect).toHaveBeenCalledWith('/not-found');
         });
+
+        it('should render error page when party URN is empty with spaces', async () => {
+          req.params.urn = '  ';
+
+          expect(res.redirect).toHaveBeenCalledWith('/not-found');
+        });
+
+        it('should submit party urn to TFM', async () => {
+          req.params.urn = '1234';
+
+          await partiesController.postPartyDetails(req, res);
+          expect(res.redirect).toHaveBeenCalledWith(`/case/${req.params._id}/parties`);
+        });
```

Comment:
Not a bloicker as PR is urgent, but i'd suggest splitting this unit test up as it's got nearly 2k lines 😮